Password optional Widget Sign In Javascript Node Guide (#3471)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/nodeexpress/integrationsteps.md

### 1: Create and display sign-in page

Create a sign-in page that captures the user's username.

<div class="half border">

![Screenshot showing a sign-in page with a username field and a Next button.](/img/pwd-optional/pwd-optional-sign-in-page.png)

</div>

> ### 2: Submit username

Reviewer: ```suggestion
### 2: The user starts the flow and submits their username
```

> When the user submits their username, call `OktaAuth.idx.authenticate()` passing in the username.

Reviewer: ```suggestion
Call `OktaAuth.idx.authenticate()` passing in the username to submit it to Identity Engine.
```

Author: Two things here:

1. I think the "When" clause is needed to give context
2. After discussions with Maurice today, I think we need to get further clarification on how to use the Identity Engine term.

Author: REMOVE WHEN ... clause

```javascript
  const { username } = req.body;
  const authClient = getAuthClient(req);
  const transaction = await authClient.idx.authenticate({ username });
```

### 3. Verify identity with the email authenticator

`OktaAuth.idx.authenticate()` returns an `IdxTransaction` object indicating that the user needs to verify their identity with the email authenticator challenge.

```json
{
  status: "PENDING",
  nextStep: {
    name: "challenge-authenticator",
    type: "email",
    authenticator: {
      type: "email",
      key: "okta_email",
      displayName: "Email",
    },
  },
}
```

> The email authenticator supports OTP and magic links, and you can integrate both methods into your application. For more information about integrating the email authenticator challenge, see the [Okta email integration guide](/docs/guides/authenticators-okta-email/nodeexpress/main/#integrate-email-challenge-with-magic-links).

Reviewer: ```suggestion
The email authenticator supports user verification by One-Time Password (OTP) and by magic links. To learn more, see the [Okta email integration guide](/docs/guides/authenticators-okta-email/nodeexpress/main/#integrate-email-challenge-with-magic-links).
```

Author: Updated in PR: https://github.com/okta/okta-developer-docs/pull/3437

> ### 4. Complete sign-in flow

Author: Updated in PR: https://github.com/okta/okta-developer-docs/pull/3437

> When the user completes the email authenticator verification, one of `OktaAuth.idx` methods returns `IdxTransaction.status` of `SUCCESS` along with access and ID tokens, which indicates a successful sign-in.

Reviewer: ```suggestion
When the user verifies their identity successfully, Identity Engine returns an `IdxTransaction.status` of `SUCCESS` along with access and ID tokens.
```

Reviewer: Also, is the reader likely to know what to do with the ID and access tokens? Also, what ID? Is there a link to cover waht to do with them? Perhaps point the user here - /docs/guides/oie-embedded-widget-use-case-basic-sign-in/nodejs/main/#_3-request-and-store-the-tokens-from-okta

Author: We've had this discussion in the past in a group forum and I remember ending with the ID and access tokens. Since returning tokens are the final step in most of these guides, we should get confirmation and clarification from other writers (Susan, etc)

Author: Saying that, I did a sentence describing the next course of action.

Author: Make it ID and Access token.
